modules/apigw/main.tf

```
@@ -4,6 +4,13 @@ resource "aws_api_gateway_rest_api" "api" {
   description              = "API for ${var.name}"
   binary_media_types       = var.binary_type
   minimum_compression_size = var.minimum_compression_size
+
+  policy = var.endpoint_type == "PRIVATE" ? data.aws_iam_policy_document.private.json : ""
```

```suggestion
  policy = var.endpoint_type == "PRIVATE" ? data.aws_iam_policy_document.private.json : null
```
Maybe `null` instead of `""`. If you pass `null` to a terraform variable it's the exact same as if you hadn't set the variable atall, so I'd be a bit more confident that it would behave the same when the `endpoint_type` isn't `PRIVATE`

---

Ah thank you, this is something I didn't know